tests/components/switch_as_x/test_siren.py

```diff
@@ -0,0 +1,117 @@
+"""Tests for the Switch as X Siren platform."""
+from homeassistant.components.siren import DOMAIN as SIREN_DOMAIN
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.components.switch_as_x.const import CONF_TARGET_DOMAIN, DOMAIN
+from homeassistant.const import (
+    CONF_ENTITY_ID,
+    SERVICE_TOGGLE,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+
+async def test_default_state(hass: HomeAssistant) -> None:
+    """Test siren switch default state."""
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={
+            CONF_ENTITY_ID: "switch.test",
+            CONF_TARGET_DOMAIN: Platform.SIREN,
+        },
+        title="Noise Maker",
+    )
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("siren.noise_maker")
+    assert state is not None
+    assert state.state == "unavailable"
+    assert state.attributes["supported_features"] == 3
+
+
+async def test_service_calls(hass: HomeAssistant) -> None:
+    """Test service calls to siren."""
```

Comment:
The test name and docstring are a bit confusing since the test tests both service calls to the siren are forwarded to the switch, and switch state changes are forwarded to the siren.

Reply:
Adjusted here: [`2ee0b6c` (#68118)](https://github.com/home-assistant/core/pull/68118/commits/2ee0b6c04c660f7d80518cf137abe9bb1d6c5144)